// @pulumi/azure-native/machinelearningservices/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.v20210401 = exports.v20210301preview = exports.v20210101 = exports.v20200901preview = exports.v20200801 = exports.v20200601 = exports.v20200515preview = exports.v20200501preview = exports.v20200401 = exports.v20200301 = exports.v20200218preview = exports.v20200101 = exports.v20191101 = exports.v20190601 = exports.v20190501 = exports.v20181119 = exports.v20180301preview = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("../utilities");
// Export members:
__exportStar(require("./getLabelingJob"), exports);
__exportStar(require("./getLinkedService"), exports);
__exportStar(require("./getLinkedWorkspace"), exports);
__exportStar(require("./getMachineLearningCompute"), exports);
__exportStar(require("./getMachineLearningDataset"), exports);
__exportStar(require("./getMachineLearningDatastore"), exports);
__exportStar(require("./getMachineLearningService"), exports);
__exportStar(require("./getPrivateEndpointConnection"), exports);
__exportStar(require("./getWorkspace"), exports);
__exportStar(require("./getWorkspaceConnection"), exports);
__exportStar(require("./labelingJob"), exports);
__exportStar(require("./linkedService"), exports);
__exportStar(require("./linkedWorkspace"), exports);
__exportStar(require("./listMachineLearningComputeKeys"), exports);
__exportStar(require("./listMachineLearningComputeNodes"), exports);
__exportStar(require("./listNotebookKeys"), exports);
__exportStar(require("./listStorageAccountKeys"), exports);
__exportStar(require("./listWorkspaceKeys"), exports);
__exportStar(require("./listWorkspaceNotebookAccessToken"), exports);
__exportStar(require("./machineLearningCompute"), exports);
__exportStar(require("./machineLearningDataset"), exports);
__exportStar(require("./machineLearningDatastore"), exports);
__exportStar(require("./machineLearningService"), exports);
__exportStar(require("./privateEndpointConnection"), exports);
__exportStar(require("./workspace"), exports);
__exportStar(require("./workspaceConnection"), exports);
// Export enums:
__exportStar(require("../types/enums/machinelearningservices"), exports);
// Export sub-modules:
const v20180301preview = require("./v20180301preview");
exports.v20180301preview = v20180301preview;
const v20181119 = require("./v20181119");
exports.v20181119 = v20181119;
const v20190501 = require("./v20190501");
exports.v20190501 = v20190501;
const v20190601 = require("./v20190601");
exports.v20190601 = v20190601;
const v20191101 = require("./v20191101");
exports.v20191101 = v20191101;
const v20200101 = require("./v20200101");
exports.v20200101 = v20200101;
const v20200218preview = require("./v20200218preview");
exports.v20200218preview = v20200218preview;
const v20200301 = require("./v20200301");
exports.v20200301 = v20200301;
const v20200401 = require("./v20200401");
exports.v20200401 = v20200401;
const v20200501preview = require("./v20200501preview");
exports.v20200501preview = v20200501preview;
const v20200515preview = require("./v20200515preview");
exports.v20200515preview = v20200515preview;
const v20200601 = require("./v20200601");
exports.v20200601 = v20200601;
const v20200801 = require("./v20200801");
exports.v20200801 = v20200801;
const v20200901preview = require("./v20200901preview");
exports.v20200901preview = v20200901preview;
const v20210101 = require("./v20210101");
exports.v20210101 = v20210101;
const v20210301preview = require("./v20210301preview");
exports.v20210301preview = v20210301preview;
const v20210401 = require("./v20210401");
exports.v20210401 = v20210401;
// Import resources to register:
const labelingJob_1 = require("./labelingJob");
const linkedService_1 = require("./linkedService");
const linkedWorkspace_1 = require("./linkedWorkspace");
const machineLearningCompute_1 = require("./machineLearningCompute");
const machineLearningDataset_1 = require("./machineLearningDataset");
const machineLearningDatastore_1 = require("./machineLearningDatastore");
const machineLearningService_1 = require("./machineLearningService");
const privateEndpointConnection_1 = require("./privateEndpointConnection");
const workspace_1 = require("./workspace");
const workspaceConnection_1 = require("./workspaceConnection");
const _module = {
    version: utilities.getVersion(),
    construct: (name, type, urn) => {
        switch (type) {
            case "azure-native:machinelearningservices:LabelingJob":
                return new labelingJob_1.LabelingJob(name, undefined, { urn });
            case "azure-native:machinelearningservices:LinkedService":
                return new linkedService_1.LinkedService(name, undefined, { urn });
            case "azure-native:machinelearningservices:LinkedWorkspace":
                return new linkedWorkspace_1.LinkedWorkspace(name, undefined, { urn });
            case "azure-native:machinelearningservices:MachineLearningCompute":
                return new machineLearningCompute_1.MachineLearningCompute(name, undefined, { urn });
            case "azure-native:machinelearningservices:MachineLearningDataset":
                return new machineLearningDataset_1.MachineLearningDataset(name, undefined, { urn });
            case "azure-native:machinelearningservices:MachineLearningDatastore":
                return new machineLearningDatastore_1.MachineLearningDatastore(name, undefined, { urn });
            case "azure-native:machinelearningservices:MachineLearningService":
                return new machineLearningService_1.MachineLearningService(name, undefined, { urn });
            case "azure-native:machinelearningservices:PrivateEndpointConnection":
                return new privateEndpointConnection_1.PrivateEndpointConnection(name, undefined, { urn });
            case "azure-native:machinelearningservices:Workspace":
                return new workspace_1.Workspace(name, undefined, { urn });
            case "azure-native:machinelearningservices:WorkspaceConnection":
                return new workspaceConnection_1.WorkspaceConnection(name, undefined, { urn });
            default:
                throw new Error(`unknown resource type ${type}`);
        }
    },
};
pulumi.runtime.registerResourceModule("azure-native", "machinelearningservices", _module);
