// @pulumi/azure-native/types/enums/timeseriesinsights/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.SkuName = exports.ReferenceDataKeyPropertyType = exports.LocalTimestampFormat = exports.EventSourceKind = exports.EnvironmentKind = exports.DataStringComparisonBehavior = exports.AccessPolicyRole = exports.v20200515 = exports.v20180815preview = exports.v20171115 = exports.v20170228preview = void 0;
// Export sub-modules:
const v20170228preview = require("./v20170228preview");
exports.v20170228preview = v20170228preview;
const v20171115 = require("./v20171115");
exports.v20171115 = v20171115;
const v20180815preview = require("./v20180815preview");
exports.v20180815preview = v20180815preview;
const v20200515 = require("./v20200515");
exports.v20200515 = v20200515;
exports.AccessPolicyRole = {
    Reader: "Reader",
    Contributor: "Contributor",
};
exports.DataStringComparisonBehavior = {
    Ordinal: "Ordinal",
    OrdinalIgnoreCase: "OrdinalIgnoreCase",
};
exports.EnvironmentKind = {
    Gen1: "Gen1",
    Gen2: "Gen2",
};
exports.EventSourceKind = {
    Microsoft_EventHub: "Microsoft.EventHub",
    Microsoft_IoTHub: "Microsoft.IoTHub",
};
exports.LocalTimestampFormat = {
    Embedded: "Embedded",
};
exports.ReferenceDataKeyPropertyType = {
    String: "String",
    Double: "Double",
    Bool: "Bool",
    DateTime: "DateTime",
};
exports.SkuName = {
    S1: "S1",
    S2: "S2",
    P1: "P1",
    L1: "L1",
};
