// @pulumi/azure-native/types/enums/hanaonazure/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.HanaProvisioningStatesEnum = exports.HanaInstanceSizeNamesEnum = exports.HanaInstancePowerStateEnum = exports.HanaHardwareTypeNamesEnum = exports.v20171103preview = void 0;
// Export sub-modules:
const v20171103preview = require("./v20171103preview");
exports.v20171103preview = v20171103preview;
exports.HanaHardwareTypeNamesEnum = {
    Cisco_UCS: "Cisco_UCS",
    HPE: "HPE",
};
exports.HanaInstancePowerStateEnum = {
    Starting: "starting",
    Started: "started",
    Stopping: "stopping",
    Stopped: "stopped",
    Restarting: "restarting",
    Unknown: "unknown",
};
exports.HanaInstanceSizeNamesEnum = {
    S72m: "S72m",
    S144m: "S144m",
    S72: "S72",
    S144: "S144",
    S192: "S192",
    S192m: "S192m",
    S192xm: "S192xm",
    S96: "S96",
    S112: "S112",
    S224: "S224",
    S224m: "S224m",
    S224om: "S224om",
    S224oo: "S224oo",
    S224oom: "S224oom",
    S224ooo: "S224ooo",
    S384: "S384",
    S384m: "S384m",
    S384xm: "S384xm",
    S384xxm: "S384xxm",
    S448: "S448",
    S448m: "S448m",
    S448om: "S448om",
    S448oo: "S448oo",
    S448oom: "S448oom",
    S448ooo: "S448ooo",
    S576m: "S576m",
    S576xm: "S576xm",
    S672: "S672",
    S672m: "S672m",
    S672om: "S672om",
    S672oo: "S672oo",
    S672oom: "S672oom",
    S672ooo: "S672ooo",
    S768: "S768",
    S768m: "S768m",
    S768xm: "S768xm",
    S896: "S896",
    S896m: "S896m",
    S896om: "S896om",
    S896oo: "S896oo",
    S896oom: "S896oom",
    S896ooo: "S896ooo",
    S960m: "S960m",
};
exports.HanaProvisioningStatesEnum = {
    Accepted: "Accepted",
    Creating: "Creating",
    Updating: "Updating",
    Failed: "Failed",
    Succeeded: "Succeeded",
    Deleting: "Deleting",
    Migrating: "Migrating",
};
