// @pulumi/azure-native/types/enums/botservice/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.SkuName = exports.Kind = exports.EnterpriseChannelState = exports.EnterpriseChannelNodeState = exports.v20210301 = exports.v20200602 = exports.v20180712 = exports.v20171201 = void 0;
// Export sub-modules:
const v20171201 = require("./v20171201");
exports.v20171201 = v20171201;
const v20180712 = require("./v20180712");
exports.v20180712 = v20180712;
const v20200602 = require("./v20200602");
exports.v20200602 = v20200602;
const v20210301 = require("./v20210301");
exports.v20210301 = v20210301;
exports.EnterpriseChannelNodeState = {
    Creating: "Creating",
    CreateFailed: "CreateFailed",
    Started: "Started",
    Starting: "Starting",
    StartFailed: "StartFailed",
    Stopped: "Stopped",
    Stopping: "Stopping",
    StopFailed: "StopFailed",
    Deleting: "Deleting",
    DeleteFailed: "DeleteFailed",
};
exports.EnterpriseChannelState = {
    Creating: "Creating",
    CreateFailed: "CreateFailed",
    Started: "Started",
    Starting: "Starting",
    StartFailed: "StartFailed",
    Stopped: "Stopped",
    Stopping: "Stopping",
    StopFailed: "StopFailed",
    Deleting: "Deleting",
    DeleteFailed: "DeleteFailed",
};
exports.Kind = {
    Sdk: "sdk",
    Designer: "designer",
    Bot: "bot",
    Function: "function",
};
exports.SkuName = {
    F0: "F0",
    S1: "S1",
};
