// @pulumi/azure-native/types/enums/resources/v20201001/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.ScriptType = exports.ResourceIdentityType = exports.OnErrorDeploymentType = exports.ManagedServiceIdentityType = exports.ExpressionEvaluationOptionsScopeType = exports.DeploymentMode = void 0;
exports.DeploymentMode = {
    Incremental: "Incremental",
    Complete: "Complete",
};
exports.ExpressionEvaluationOptionsScopeType = {
    NotSpecified: "NotSpecified",
    Outer: "Outer",
    Inner: "Inner",
};
exports.ManagedServiceIdentityType = {
    UserAssigned: "UserAssigned",
};
exports.OnErrorDeploymentType = {
    LastSuccessful: "LastSuccessful",
    SpecificDeployment: "SpecificDeployment",
};
exports.ResourceIdentityType = {
    SystemAssigned: "SystemAssigned",
    UserAssigned: "UserAssigned",
    SystemAssigned_UserAssigned: "SystemAssigned, UserAssigned",
    None: "None",
};
exports.ScriptType = {
    AzurePowerShell: "AzurePowerShell",
    AzureCLI: "AzureCLI",
};
