// @pulumi/azure-native/config/vars.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.useMsi = exports.tenantId = exports.subscriptionId = exports.partnerId = exports.msiEndpoint = exports.environment = exports.disablePulumiPartnerId = exports.clientSecret = exports.clientId = exports.clientCertificatePath = exports.clientCertificatePassword = exports.auxiliaryTenantIds = void 0;
const pulumi = require("@pulumi/pulumi");
let __config = new pulumi.Config("azure-native");
exports.auxiliaryTenantIds = __config.getObject("auxiliaryTenantIds");
/**
 * The password associated with the Client Certificate. For use when authenticating as a Service Principal using a Client Certificate
 */
exports.clientCertificatePassword = __config.get("clientCertificatePassword");
/**
 * The path to the Client Certificate associated with the Service Principal for use when authenticating as a Service Principal using a Client Certificate.
 */
exports.clientCertificatePath = __config.get("clientCertificatePath");
/**
 * The Client ID which should be used.
 */
exports.clientId = __config.get("clientId");
/**
 * The Client Secret which should be used. For use When authenticating as a Service Principal using a Client Secret.
 */
exports.clientSecret = __config.get("clientSecret");
/**
 * This will disable the Pulumi Partner ID which is used if a custom `partnerId` isn't specified.
 */
exports.disablePulumiPartnerId = __config.getObject("disablePulumiPartnerId");
/**
 * The Cloud Environment which should be used. Possible values are public, usgovernment, german, and china. Defaults to public.
 */
exports.environment = __config.get("environment");
/**
 * The path to a custom endpoint for Managed Service Identity - in most circumstances this should be detected automatically.
 */
exports.msiEndpoint = __config.get("msiEndpoint");
/**
 * A GUID/UUID that is registered with Microsoft to facilitate partner resource usage attribution.
 */
exports.partnerId = __config.get("partnerId");
/**
 * The Subscription ID which should be used.
 */
exports.subscriptionId = __config.get("subscriptionId");
/**
 * The Tenant ID which should be used.
 */
exports.tenantId = __config.get("tenantId");
/**
 * Allowed Managed Service Identity be used for Authentication.
 */
exports.useMsi = __config.getObject("useMsi");
