// @pulumi/azure-native/types/enums/edgeorder/v20201201preview/index.js
"use strict";
// *** WARNING: this file was generated by the Pulumi SDK Generator. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.TransportShipmentTypes = exports.SupportedFilterTypes = exports.OrderItemType = exports.NotificationStageName = exports.DoubleEncryptionStatus = exports.AddressType = void 0;
exports.AddressType = {
    /**
     * Address type not known.
     */
    None: "None",
    /**
     * Residential Address.
     */
    Residential: "Residential",
    /**
     * Commercial Address.
     */
    Commercial: "Commercial",
};
exports.DoubleEncryptionStatus = {
    /**
     * Double encryption is disabled
     */
    Disabled: "Disabled",
    /**
     * Double encryption is enabled
     */
    Enabled: "Enabled",
};
exports.NotificationStageName = {
    /**
     * Notification at device prepared stage.
     */
    DevicePrepared: "DevicePrepared",
    /**
     * Notification at device dispatched stage.
     */
    Shipped: "Shipped",
    /**
     * Notification at device delivered stage.
     */
    Delivered: "Delivered",
    /**
     * Notification at device picked up from user stage.
     */
    PickedUp: "PickedUp",
    /**
     * Notification at device received at Azure datacenter stage.
     */
    AtAzureDC: "AtAzureDC",
    /**
     * Notification at data copy started stage.
     */
    DataCopy: "DataCopy",
};
exports.OrderItemType = {
    /**
     * Purchase OrderItem.
     */
    Purchase: "Purchase",
    /**
     * Rental OrderItem.
     */
    Rental: "Rental",
};
exports.SupportedFilterTypes = {
    /**
     * Ship to country
     */
    ShipToCountries: "ShipToCountries",
    /**
     * Double encryption status
     */
    DoubleEncryptionStatus: "DoubleEncryptionStatus",
};
exports.TransportShipmentTypes = {
    /**
     * Shipment Logistics is handled by the customer.
     */
    CustomerManaged: "CustomerManaged",
    /**
     * Shipment Logistics is handled by Microsoft.
     */
    MicrosoftManaged: "MicrosoftManaged",
};
